enterprise/backend/src/metabase_enterprise/serialization/dump.clj

```diff
@@ -34,8 +35,11 @@
   (io/make-parents filename)
   (spit filename (yaml/generate-string obj :dumper-options {:flow-style :block})))
 
-(def ^:private as-file?
-  (comp (set (map type [Pulse Dashboard Metric Segment Field User])) type))
+(defn- as-file?
+  [instance]
+  (some (fn [model]
+          (mi/instance-of? model instance))
+        [Pulse Dashboard Metric Segment Field User]))
```

Comment:
This was checking whether instances were of a `type` like `DashboardInstance`, which doesn't work in Toucan 2. Updated to use the `mi/instance-of?` helper function I added recently which works with either Toucan 1 or Toucan 2